be/src/vec/functions/like.cpp

```diff
@@ -372,10 +372,12 @@ Status FunctionLikeBase::execute_impl(FunctionContext* context, Block& block,
         const auto pattern_col = block.get_by_position(arguments[1]).column;
 
         if (const auto* str_patterns = check_and_get_column<ColumnString>(pattern_col.get())) {
-            DCHECK_EQ(str_patterns->size(), 1);
-            const auto& pattern_val = str_patterns->get_data_at(0);
-            RETURN_IF_ERROR(vector_const(*values, &pattern_val, vec_res, state->function,
-                                         &state->search_state));
+            for (int i = 0; i < input_rows_count; i++) {
```

Comment:
should we forbid the pattern column is non-const?

Reply:
I test mysql. mysql support the query non-const way